core/ecschema-metadata/src/Deserialization/Helper.ts

```diff
@@ -142,6 +143,15 @@ export class SchemaReadHelper<T = unknown> {
     return schemaItem !== undefined;
   }
 
+  /**
+   * Same as isSchemaItemLoaded, but also asserts that the schema item is defined using a type predicate.
+   * @note isSchemaItemLoaded already checks if the item is defined, but updating it to be a type predicate
+   * would break existing code that uses it.
+   */
+  protected isSchemaItemLoadedAndDefined(schemaItem: SchemaItem | undefined): schemaItem is SchemaItem {
```

Comment:
why is it breaking to make it a type predicate?

Reply:
Done.